dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -100,4 +100,39 @@ def manual_pass
     flash[:alert] = "UserLevel (ID: #{user_level.id}) updated"
     redirect_to :manual_pass_form
   end
+
+  def permissions_form
+  end
+
+  # Grants the indicated permission to the indicated user. Expects params[:email] and
+  # params[:permission] to be populated.
+  def grant_permission
+    permission = params[:permission]
+    user = User.find_by_email_or_hashed_email params[:email]
+
+    unless user && user.teacher?
+      flash[:alert] = "FAILED: user #{params[:email]} could not be found or was not a teacher"
+      redirect_to :permissions_form
+      return
+    end
+
+    if permission == 'admin'
+      user.update!(admin: true)
+      flash[:alert] = "User #{user.id} granted admin status"
+    else
+      user.permission = permission
+      flash[:alert] = "User #{user.id} granted #{permission} permission"
+    end
+    redirect_to :permissions_form
+  end
+
+  def revoke_all_permissions
```

Comment:
Note that, though I've added tests to the model, I'm being lazy and testing for effect in controller (not updating tests) rather than confirming `user.revoke_all_privileges` is being called.